examples/gno.land/r/gov/dao/v2/render.gno

```diff
@@ -0,0 +1,122 @@
+package govdao
+
+import (
+	"strconv"
+	"strings"
+
+	"gno.land/p/demo/dao"
+	"gno.land/p/demo/ufmt"
+	"gno.land/p/moul/txlink"
+	"gno.land/r/demo/users"
+)
+
+func Render(path string) string {
+	var out string
+
+	if path == "" {
+		out += "# GovDAO Proposals\n\n"
+		numProposals := d.Size()
+
+		if numProposals == 0 {
+			out += "No proposals found :(" // corner case
+			return out
+		}
+
+		offset := uint64(0)
+		if numProposals >= 10 {
+			offset = uint64(numProposals) - 10
+		}
+
+		// Fetch the last 10 proposals
+		proposals := d.Proposals(offset, uint64(10))
+		for i := len(proposals) - 1; i >= 0; i-- {
+			prop := proposals[i]
+
+			title := prop.Title()
+			if len(title) > 40 {
+				title = title[:40] + "..."
+			}
+
+			propID := offset + uint64(i)
+			out += ufmt.Sprintf("## [Prop #%d - %s](/r/gov/dao/v2:%d)\n\n", propID, title, propID)
+			out += ufmt.Sprintf("#### Status: %s\n\n", strings.ToUpper(prop.Status().String()))
+
+			user := users.GetUserByAddress(prop.Author())
+			authorDisplayText := prop.Author().String()
+			if user != nil {
+				authorDisplayText = ufmt.Sprintf("[%s](/r/demo/users:%s)", user.Name, user.Name)
+			}
+
+			out += ufmt.Sprintf("#### Author: %s\n\n", authorDisplayText)
+
+			if i != 0 {
+				out += "---\n\n"
+			}
+		}
+
+		return out
+	}
+
+	// Display the detailed proposal
+	idx, err := strconv.Atoi(path)
+	if err != nil {
+		return "404: Invalid proposal ID"
+	}
+
+	// Fetch the proposal
+	prop, err := d.ProposalByID(uint64(idx))
+	if err != nil {
+		return ufmt.Sprintf("unable to fetch proposal, %s", err.Error())
+	}
+
+	// Render the proposal page
+	out += renderPropPage(prop, idx)
+
+	return out.String()
+}
+
+func renderPropPage(prop dao.Proposal, idx int) string {
+	var out string
+
+	out += ufmt.Sprintf("# Proposal #%d - %s\n\n", idx, prop.Title())
+	out += prop.Render()
```

Comment:
rendering the author before this would improve clarity and make the information more straightforward.

Reply:
I think the prop page looks better if the author is at the bottom.

This is how it is when it's on the top.

<img width="894" alt="Screenshot 2024-11-21 at 12 53 44" src="https://github.com/user-attachments/assets/92ac23cc-bd54-4e60-a964-2e2976a033b9">

Reply:
My suggestion is based on the idea that for a proposal, the order of title, author, desc, and details would make it clearer. 

I would want to immediately see the title and the corresponding author at first glance. For example, if it's a new proposal to join the valset, I’d like to know right away who it’s from (even if it’s just an address).

(Personal opinion).

Reply:
> For example, if it's a new proposal to join the valset, I’d like to know right away who it’s from (even if it’s just an address).

I think that in this case, new validator address and author don't necessarily have to match, no?

Reply:
just something similar in my mind, but yeah I'm  not insisting on this :)

<img width="734" alt="image" src="https://github.com/user-attachments/assets/bd183103-5507-415b-9e1b-7384fe620d9f">